e2e/test/PnpServiceTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    [TestCategory("PlugAndPlay")]
+    public class PnpServiceTests
+    {
+        private const string DevicePrefix = "plugAndPlayDevice";
+        private const string ModulePrefix = "plugAndPlayModule";
+        private const string TestModelId = "dtmi:com:example:testModel;1";
+
+        [TestMethod]
+        public async Task DeviceTwin_Contains_ModelId()
+        {
+            // Setup
+
+            // Create a device.
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix).ConfigureAwait(false);
+            // Send model ID with MQTT connect packet to make the device plug and play.
+            var options = new ClientOptions
+            {
+                ModelId = TestModelId,
+            };
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, Client.TransportType.Mqtt_Tcp_Only, options);
+            await deviceClient.OpenAsync().ConfigureAwait(false);
+
+            // Act
+
+            // Get device twin.
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            Twin twin = await registryManager.GetTwinAsync(testDevice.Device.Id).ConfigureAwait(false);
+
+            // Assert
+            twin.ModelId.Should().Be(TestModelId, "because the device was created as plug and play");
+
+            // Cleanup
+            await registryManager.RemoveDeviceAsync(testDevice.Id).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task ModuleTwin_Contains_ModelId()
+        {
+            // Setup
+
+            // Create a module.
+            TestModule testModule = await TestModule.GetTestModuleAsync(DevicePrefix, ModulePrefix).ConfigureAwait(false);
+            // Send model ID with MQTT connect packet to make the module plug and play.
+            var options = new ClientOptions
+            {
+                ModelId = TestModelId,
+            };
+            using var moduleClient = ModuleClient.CreateFromConnectionString(testModule.ConnectionString, Client.TransportType.Mqtt_Tcp_Only, options);
```

Comment:
is there way to not hard code this?

Reply:
I have not found a way but we can extend this test in the future if it is needed. For now it anyway only works with MQTT.